assets/src/components/multiselect-control/index.js

```diff
@@ -3,19 +3,13 @@ import './editor.scss';
 import wp from 'wp';
 import Select from 'react-select';
 
-const {
-	useState,
-	useEffect,
-} = wp.element;
 const { BaseControl } = wp.components;
 
 export default function MultiSelectControl( { value = [], options = [], label, help, placeholder, onChange, ...reactSelectProps } ) {
-	const [ values, setValues ] = useState( [] );
-
-	useEffect(
-		() => setValues( options.filter( ( option ) => value.includes( option.value ) ) ),
-		[ value ]
-	);
+	// Filter out any values that are not in the options list and map them to the option object.
+	const values = value
+		.filter( token => options.some( option => option.value === token ) )
```

Comment:
@abhi3315 can we follow WPCS here - parentheses around params: `.filter( ( token ) =>`. Can you update everywhere in this PR?